spark_log_parser/parsing_models/application_model_v2.py

```diff
@@ -580,175 +836,118 @@ def getAllDriverAccumData(self, appobj):
-        if hasattr(self, "stageData"):
-            saveDat["stageData"] = self.stageData.reset_index().to_dict("list")
-        if hasattr(self, "taskData"):
-            saveDat["taskData"] = self.taskData.reset_index().to_dict("list")
-        if hasattr(self, "accumData"):
-            saveDat["accumData"] = self.accumData.reset_index().to_dict("list")
-        if self.existsSQL:
-            saveDat["sqlData"] = self.sqlData.reset_index().to_dict("list")
-        if self.existsExecutors:
-            saveDat["executors"] = self.executorData.reset_index().to_dict("list")
-
-        saveDat["metadata"] = self.metadata
+        if spark_app.existsSQL:
+            trecent = []
+            for qid in spark_app.sqlData.index.values:
+                tstart = spark_app.sqlData.loc[qid]["start_time"]
 
-        if (filepath is not None) and ("s3://" in filepath):
-            self.save_to_s3(saveDat, filepath, compress)
-        else:
-            self.save_to_local(saveDat, filepath, compress)
+                tmp = tcomp[tcomp < tstart]
 
-    def save_to_local(self, saveDat, filepath, compress):
-        if filepath is None:
-            if self.spark_eventlog_path is None:
-                raise Exception('No input eventlog found. Must specify "filepath".')
-            inputFile = os.path.basename(os.path.normpath(self.spark_eventlog_path)).replace(
-                ".gz", ""
-            )
-            filepath = inputFile + "-sync"
+                if len(tmp) == 0:
+                    trecent.append(0)
+                else:
+                    trecent.append(tstart - tcomp[tcomp < tstart].max())
+            spark_app.sqlData["time_since_last_event"] = trecent
 
-        if compress is False:
-            with open(filepath + ".json", "w") as fout:
-                fout.write(json.dumps(saveDat))
-        elif compress is True:
-            with gzip.open(filepath + ".json.gz", "w") as fout:
-                fout.write(json.dumps(saveDat).encode("ascii"))
-        logging.info("Saved object locally to: %s" % (filepath))
+        return spark_app
 
-    def save_to_s3(self, saveDat, filepath, compress):
-        s3 = boto3.client("s3")
 
-        # Extract bucket and key from s3 filepath
-        path = filepath.replace("s3://", "").split("/")
-        bucket = path[0]
-        key = ("/".join(path[1:])).lstrip("/") + ".json"
+"""
+TODO - this function maintains the filepath inputs in order to make migration easier. This should be
+removed at some point.
+"""
 
-        if compress is False:
-            s3.put_object(Bucket=bucket, Body=json.dumps(saveDat).encode("utf-8"), Key=key)
-        else:
-            dat = gzip.compress(json.dumps(saveDat).encode("utf-8"))
-            s3.put_object(Bucket=bucket, Body=dat, Key=key + ".gz")
 
-        logging.info("Saved object to cloud: %s" % (key))
+async def create_spark_application_async(
+    spark_eventlog_parsed_path=None,
+    spark_eventlog_path=None,
```

Comment:
Might be a good opportunity to clean up this interface: single required event log arg and a flag indicating whether it's parsed?

Reply:
haha, yeah, I agree. I was planning on actually just removing this in a PR or two, but happy to make this change as well.

Reply:
Yeah, not a top concern obviously, Was just thinking by tightening it up it's harder to mistake how the function should be called.